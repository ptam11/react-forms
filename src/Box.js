import React, { Component } from 'react';

class Box extends Component {
  static defaultProps = {
    width: 100,
    height: 100,
    backgroundColor: 'dodgerblue'
  }
  
  render() {
    const style = {
      width: `${this.props.width}px`,
      height: `${this.props.height}px`,
      backgroundColor: `${this.props.backgroundColor}`
    }
    return (
      <div style={style} className="Box" />
    );
  }

}

export default Box;
