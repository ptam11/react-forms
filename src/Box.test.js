import React from 'react';
import Box from './Box';
import { shallow, mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({
  adapter: new Adapter()
});

it('renders without errors', () => {
  shallow(<Box />);
});

it('snapshot matches', () => {
  let wrapper = shallow(<Box />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

xit('it renders initial message', () => {
  let wrapper = mount(<Box />);
  expect(wrapper.exists('p')).toEqual(true);
  expect(wrapper.html()).toContain("Think of a Question");
});

xit('it renders correct message', () => {
  let wrapper = mount(<Box />);
  wrapper.setState({ msg: "Hello", color: "rebeccapurple" });
  let p_elem = wrapper.find('p').first();
  expect(p_elem.equals(<p>Hello</p>)).toEqual(true);
});

const shortList = [
  { msg: "Whiskey is a dog.", color: "green" },
  { msg: "Whiskey likes food.", color: "green" },
  { msg: "Whiskey likes sleeping.", color: "green" },
]

xit('it changes default message on click', () => {
  let wrapper = mount(<Box  answers={shortList}/>);
  wrapper.simulate("click");
  expect(wrapper.html()).toContain("Whiskey");
});