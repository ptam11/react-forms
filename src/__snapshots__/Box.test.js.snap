// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches 1`] = `
<div
  className="Box"
  style={
    Object {
      "backgroundColor": "dodgerblue",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
